qemu/tests/hugepage_reset.py

```diff
@@ -21,6 +21,33 @@ def run(test, params, env):
     :param env: Dictionary with the test environment.
     """
 
+    def allocate_largepages_per_node():
+        """
+        The functions is inteded to set 1G hugepages per NUMA
+        node when the system has four or more of these nodes.
+        For this function to work the hugepage size should be 1G.
+        This way a QEMU failure could be avoided if it's unable
+        to allocate memory.
+        """
+        node_list = host_numa_node.online_nodes_withcpumem
+        if len(node_list) >= 4:
```

Comment:
If nodes < 4, we don't need to do anything? Means we don't need to setup hugepages for test?

Reply:
For this particular test case, it's expected by default 1G hugepages: https://github.com/autotest/tp-qemu/blob/157fd7c78b8862dde41a31f02f193b6de5d170e3/qemu/tests/cfg/hugepage_reset.cfg#L9

So as far as I know, **automatic** 1G hugepage setting is **not** **supported** by avocado-vt, so them have to be set manually, (it's also described in the test cfg).

Now going back to this function, the `mem` param is set to 4G, which means that in the `on_numa_node` variant 4G hugepages are needed at least in one node. For systems with more than 4 NUMA nodes, problems may arise because of a user not being aware of the system's NUMA topology can do an insufficient 1G hugepages allocation. So the idea of the function is trying to **help** setting the needed hugepages if possible.

Reply:
Correcting myself: automation support (in part) 1G hugepage setting, so at the end not needed to adapt the test for different NUMA topologies just fix the `node_mem_free` and `mem` comparison. The user will still need to set the hugepage size in the kernel command line